microsoft/vscode
# compress schemas by using $refs for duplicates

For https://github.com/microsoft/vscode/issues/214205


Compress the JSON schemas that we use for our configuration files by using `$refs` for duplicate subtrees.

To test:
- start with no editors open
- run `Set Debug Level...` to `Debug`
- open `package.json`

Look at the `Window` output view to see the JSON schemas that are generated, and the compression rate.

2024-07-03 21:08:53.976 [debug] /vscode-extensions: 574046 -> 226267 (61%) Took 14ms
2024-07-03 21:08:54.142 [debug] /settings/configurationDefaults: 1192405 -> 817228 (31%) Took 38ms
2024-07-03 21:08:54.274 [debug] /settings/resourceLanguage: 124234 -> 120206 (3%) Took 4ms
2024-07-03 21:08:54.346 [debug] /launch: 1879294 -> 211792 (89%) Took 49ms
2024-07-03 21:08:54.358 [debug] /token-styling: 20686 -> 18176 (12%) Took 2ms
2024-07-03 21:08:54.361 [debug] /textmate-colors: 5784 -> 3949 (32%) Took 0ms
2024-07-03 21:08:54.371 [debug] /workbench-colors: 246988 -> 204613 (17%) Took 7ms
